### ISSUE TITLE
warning: already initialized constant Mobility::Loaded::Rails

### ISSUE THREAD

**Issue by jnylen** [shioyama/mobility#75]
Just some warnings popping up about Rails on Hanami + Sequel.
Hanami 1.0.0
Sequel 4.*
Mobility 0.2.0

/home/vagrant/.rvm/gems/ruby-2.3.3/gems/mobility-0.2.0/lib/mobility.rb:57: warning: already initialized constant Mobility::Loaded::Rails
/home/vagrant/.rvm/gems/ruby-2.3.3/gems/mobility-0.2.0/lib/mobility.rb:53: warning: previous definition of Rails was here

**Comment by jnylen** [comment]
@shioyama,
This also seems to be happening:
/home/vagrant/.rvm/gems/ruby-2.3.3/bundler/gems/mobility-d58081ef2e2f/lib/mobility/backends/sequel/pg_hash.rb:42:in `block in <class:PgHash>': uninitialized constant Mobility::Sequel::ColumnChanges (NameError)

**Comment by shioyama** [comment]
Thanks, about the first warning, I'm guessing that Mobility is able to load 'rails' but then something fails to load after that, probably 'rails/generators'. The result would be that it sets `Mobility::Loaded::Rails` first to true, then back to false.

I can't imagine that rails would be included in your application... does `require 'rails'` raise a `LoadError` from console with your Gemfile?

**Comment by jnylen** [comment]
Hi @shioyama,
It raises:
=> false

I have these "rails" gems in the lockfile:
rails-dom-testing
railties
rails-html-sanitizer

Any idea how to fix the Mobility::Sequel::ColumnChanges one?

**Comment by shioyama** [comment]
Ok, then I see the issue. The core problem is that ruby doesn't have "optional dependencies", so the way gems typically do this is to try requiring a dependency, and catch the `LoadError` if it fails to load. But in this case you have rails, but not another dependency, so this breaks down.

I can this I think, I'll just need to require the generators separately and not assume they are bundled with rails.

The second issue is just a missing requires in "mobility/backends/sequel/pg_hash". Can you try pointing to the `require_column_changes` [branch](https://github.com/shioyama/mobility/tree/require_column_changes)?

**Comment by shioyama** [comment]
I've merged the branch into `master`, if you use the latest the ColumnChanges issue should be fixed.

**Comment by shioyama** [comment]
Also, I realized the other warning issue is because in your case you have "rails", but not "activerecord", whereas the rails generators assume that AR is loaded. This is an issue of how things are required, I'll fix it in the next day or two and release a new version. Thanks for pointing out the issue!

Let me know if the other (second) issue is fixed on `master`.

**Comment by shioyama** [comment]
Try the `load_rails` [branch](https://github.com/shioyama/mobility/tree/load_rails) which has fixes for both issues.

**Comment by jnylen** [comment]
@shioyama,
load_rails fixed both.

But a 2 new errors occur. If I dont add data to a jsonb that was created with:
`jsonb :synopsis`

It will say: `Sequel::Error: The AND operator requires at least 1 argument`

If I add data to the only 2 jsonb columns in the model (Sequel::Model) it gives.
``
2.3.3 :004 > 
m.synopsis = "test"
 => "test" 
2.3.3 :005 > m.save
Sequel::DatabaseError: PG::DatatypeMismatch: ERROR:  column "synopsis" is of type jsonb but expression is of type boolean
LINE 1: ...gas" ("synopsis", "title", "created_at") VALUES (('en' = 'te...
                                                             ^
HINT:  You will need to rewrite or cast the expression.
``

**Comment by shioyama** [comment]
@jnylen Are you sure the column is of type `jsonb`? What does your schema look like?

**Comment by shioyama** [comment]
Oh never mind...

**Comment by jnylen** [comment]
By the way,
Here are the sequel related gems in Gemfile:
```
    sequel (4.49.0)
    sequel-annotate (1.0.0)
      sequel (>= 4)
    sequel-rake (0.1.0)
      thor (~> 0.19)
    sequel_pg (1.7.0)
      pg (>= 0.8.0)
      sequel (>= 4.0.0)
    sequel_polymorphic (0.3.1)
      sequel (~> 4)
    sequel_sluggable (0.0.6)
      sequel (>= 3.0.0)
```

**Comment by shioyama** [comment]
I'm going to close this issue and merge the branch since that resolves it. Can you please post a new issue about the jsonb error?

**Comment by shioyama** [comment]
Released 0.2.1 with these fixes: https://rubygems.org/gems/mobility/versions/0.2.1
